gobblin-runtime/src/main/java/gobblin/runtime/LimitingExtractorDecorator.java

```diff
@@ -0,0 +1,86 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime;
+
+import java.io.IOException;
+
+import gobblin.source.extractor.DataRecordException;
+import gobblin.source.extractor.Extractor;
+
+
+/**
+ * A decorator class for {@link Extractor} that uses a {@link Limiter} on data record extraction.
+ *
+ * <p>
+ *   The fact that the {@link Limiter} is passed in as a parameter to the constructor
+ *   {@link LimitingExtractorDecorator#LimitingExtractorDecorator(Extractor, Limiter)}
+ *   means multiple {@link LimitingExtractorDecorator}s can share a single {@link Limiter}
+ *   or each individual {@link LimitingExtractorDecorator} has its own {@link Limiter}.
+ *   The first case is useful for throttling at above the task level, e.g., at the job level.
+ * </p>
+ *
+ * @param <S> output schema type
+ * @param <D> output record type
+ *
+ * @author ynli
+ */
+public class LimitingExtractorDecorator<S, D> implements Extractor<S, D> {
```

Comment:
I think it also makes sense to divide the `Limiter`s into two groups, `NonRefillableLimiter` and `RefillableLimiter`, and `LimitingExtractorDecorator` should only accept `NonRefillableLimiter`.